### ISSUE TITLE
Check Wazuh compatibility with new version Amazon Linux 2023 2023.3.20231218

### ISSUE THREAD

**Issue by mjcr99** [wazuh/wazuh#20991]
# Description
Hello team, this issue is to check the full compatibility of Wazuh on the new found version of Amazon Linux 2023 2023.3.20231218 operating system.

OSs checks issue: #20989

For this, it is necessary to perform the following tests to check that everything works as expected:

- [x] Agent and server (if possible) instalations.
- [x] O.S. reporting in the interface.
- [x] WPK installation / upgrade / uninstall / upgrade.
- [x] Enrollment and connectivity with the manager.
- [x] FIM: Real-time and who-data engines (if available).
- [x] SCA: Policy support.
- [x] Vulnerability Detector: Vulnerability support.
- [x] Syscollector: Complete inventory.
- [x] Active Response: port blocking.
- [x] Log capture (Eventchannel for Windows and ULS for macOS).

**Comment by ncvicchi** [comment]
Blocked until AL2023 VM is provided.

**Comment by ncvicchi** [comment]
Machine was provided so testings began

**Comment by ncvicchi** [comment]
# Testing

###   :green_circle:  Agent and server, (enrollment and connectivity with the manager)

Agent installed at the Amazon Lnux 2023 2023.3.20231218
 machine by following [this guide](https://documentation.wazuh.com/current/installation-guide/wazuh-agent/wazuh-agent-package-linux.html).

The manager was implemented by running the OVA 4.7.1 on a local VM.

<details><summary>Dashboard</summary>

![image](https://github.com/wazuh/internal-devel-requests/assets/69121070/706accb7-084e-4a6b-ba58-12011dab8c87)

</details>

<details><summary>Agent's log</summary>

```console
[root@ip-172-31-33-67 ec2-user]# cat /var/ossec/logs/ossec.log
2023/12/26 18:41:03 wazuh-execd: INFO: Started (pid: 8265).
2023/12/26 18:41:04 wazuh-agentd: INFO: (1410): Reading authentication keys file.
2023/12/26 18:41:04 wazuh-agentd: INFO: Using notify time: 10 and max time to reconnect: 60
2023/12/26 18:41:04 wazuh-agentd: INFO: Version detected -> Linux |ip-172-31-33-67.ec2.internal |6.1.66-91.160.amzn2023.x86_64 |#1 SMP PREEMPT_DYNAMIC Wed Dec 13 04:50:24 UTC 2023 |x86_64 [Amazon Linux|amzn: 2023] - Wazuh v4.7.1
2023/12/26 18:41:04 wazuh-agentd: INFO: Started (pid: 8283).
2023/12/26 18:41:04 wazuh-agentd: INFO: Requesting a key from server: 152.168.76.36
2023/12/26 18:41:04 wazuh-agentd: INFO: No authentication password provided
2023/12/26 18:41:04 wazuh-agentd: INFO: Using agent name as: ip-172-31-33-67.ec2.internal
2023/12/26 18:41:04 wazuh-agentd: INFO: Waiting for server reply
2023/12/26 18:41:04 wazuh-agentd: INFO: Valid key received
2023/12/26 18:41:04 wazuh-agentd: INFO: Waiting 20 seconds before server connection
...
2023/12/26 18:41:24 wazuh-agentd: INFO: (1410): Reading authentication keys file.
2023/12/26 18:41:24 wazuh-agentd: INFO: Using AES as encryption method.
2023/12/26 18:41:24 wazuh-agentd: INFO: Trying to connect to server ([152.168.76.36]:1514/tcp).
2023/12/26 18:41:25 wazuh-agentd: INFO: (4102): Connected to the server ([152.168.76.36]:1514/tcp).
...
2023/12/26 18:41:26 wazuh-modulesd: INFO: Agent is now online. Process unlocked, continuing...
2023/12/26 18:41:27 wazuh-agentd: INFO: Agent is restarting due to shared configuration changes.
...
2023/12/26 18:41:28 wazuh-agentd: INFO: (1225): SIGNAL [(15)-(Terminated)] Received. Exit Cleaning...
2023/12/26 18:41:28 wazuh-execd: INFO: (1314): Shutdown received. Deleting responses.
2023/12/26 18:41:28 wazuh-execd: INFO: (1225): SIGNAL [(15)-(Terminated)] Received. Exit Cleaning...
2023/12/26 18:41:29 wazuh-execd: INFO: Started (pid: 8596).
2023/12/26 18:41:30 wazuh-agentd: INFO: (1410): Reading authentication keys file.
2023/12/26 18:41:30 wazuh-agentd: INFO: Using notify time: 10 and max time to reconnect: 60
2023/12/26 18:41:30 wazuh-agentd: INFO: Version detected -> Linux |ip-172-31-33-67.ec2.internal |6.1.66-91.160.amzn2023.x86_64 |#1 SMP PREEMPT_DYNAMIC Wed Dec 13 04:50:24 UTC 2023 |x86_64 [Amazon Linux|amzn: 2023] - Wazuh v4.7.1
2023/12/26 18:41:30 wazuh-agentd: INFO: Started (pid: 8614).
2023/12/26 18:41:30 wazuh-agentd: INFO: Using AES as encryption method.
2023/12/26 18:41:30 wazuh-agentd: INFO: Trying to connect to server ([152.168.76.36]:1514/tcp).
2023/12/26 18:41:31 wazuh-agentd: INFO: (4102): Connected to the server ([152.168.76.36]:1514/tcp).
...
2023/12/26 18:41:04 wazuh-agentd: INFO: (1410): Reading authentication keys file.
2023/12/26 18:41:04 wazuh-agentd: INFO: No authentication password provided
2023/12/26 18:41:24 wazuh-agentd: INFO: (1410): Reading authentication keys file.
2023/12/26 18:41:30 wazuh-agentd: INFO: (1410): Reading authentication keys file.

```

</details>

<details><summary>wazuh-control status before reboot</summary>

```console

[root@wazuh-server wazuh-user]# /var/ossec/bin/wazuh-control status
wazuh-clusterd not running...
wazuh-modulesd is running...
wazuh-monitord is running...
wazuh-logcollector is running...
wazuh-remoted is running...
wazuh-syscheckd is running...
wazuh-analysisd is running...
wazuh-maild not running...
wazuh-execd is running...
wazuh-db is running...
wazuh-authd is running...
wazuh-agentlessd not running...
wazuh-integratord not running...
wazuh-dbd not running...
wazuh-csyslogd not running...
wazuh-apid is running...

```
</details>

<details><summary>wazuh-control status after reboot</summary>

```console

[root@wazuh-server wazuh-user]# /var/ossec/bin/wazuh-control status
wazuh-clusterd not running...
wazuh-modulesd is running...
wazuh-monitord is running...
wazuh-logcollector is running...
wazuh-remoted is running...
wazuh-syscheckd is running...
wazuh-analysisd is running...
wazuh-maild not running...
wazuh-execd is running...
wazuh-db is running...
wazuh-authd is running...
wazuh-agentlessd not running...
wazuh-integratord not running...
wazuh-dbd not running...
wazuh-csyslogd not running...
wazuh-apid is running...

```

</details>

<details><summary>Manager log</summary>

```console
2023/12/26 18:18:55 wazuh-authd: INFO: Started (pid: 11322).
2023/12/26 18:18:55 wazuh-authd: INFO: Accepting connections on port 1515. No password required.
2023/12/26 18:18:55 wazuh-authd: INFO: Setting network timeout to 1.000000 sec.
2023/12/26 18:41:04 wazuh-authd: INFO: New connection from 54.198.50.96
2023/12/26 18:41:04 wazuh-authd: INFO: Received request for a new agent (ip-172-31-33-67.ec2.internal) from: 54.198.50.96
2023/12/26 18:41:04 wazuh-authd: INFO: Agent key generated for 'ip-172-31-33-67.ec2.internal' (requested by any)

```
</details>

###  :green_circle: FIM scheduled

Using default configuration (just changed frequency to 60 seconds):

<details><summary>Results</summary>

<details><summary>Manager's log></summary>

```console

** Alert 1703618041.23510: - ossec,syscheck,syscheck_entry_added,syscheck_file,pci_dss_11.5,gpg13_4.11,gdpr_II_5.1.f,hipaa_164.312.c.1,hipaa_164.312.c.2,nist_800_53_SI.7,tsc_PI1.4,tsc_PI1.5,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,
2023 Dec 26 19:14:01 (ip-172-31-33-67.ec2.internal) any->syscheck
Rule: 554 (level 5) -> 'File added to the system.'
File '/etc/test1.txt' added
Mode: scheduled

Attributes:
 - Size: 0
 - Permissions: rw-r--r--
 - Date: Tue Dec 26 19:13:41 2023
 - Inode: 4638
 - User: root (0)
 - Group: root (0)
 - MD5: d41d8cd98f00b204e9800998ecf8427e
 - SHA1: da39a3ee5e6b4b0d3255bfef95601890afd80709
 - SHA256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855

** Alert 1703618105.24206: - ossec,syscheck,syscheck_entry_modified,syscheck_file,pci_dss_11.5,gpg13_4.11,gdpr_II_5.1.f,hipaa_164.312.c.1,hipaa_164.312.c.2,nist_800_53_SI.7,tsc_PI1.4,tsc_PI1.5,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,
2023 Dec 26 19:15:05 (ip-172-31-33-67.ec2.internal) any->syscheck
Rule: 550 (level 7) -> 'Integrity checksum changed.'
File '/etc/test1.txt' modified
Mode: scheduled
Changed attributes: size,mtime,md5,sha1,sha256
Size changed from '0' to '5'
Old modification time was: '1703618021', now it is '1703618071'
Old md5sum was: 'd41d8cd98f00b204e9800998ecf8427e'
New md5sum is : 'd8e8fca2dc0f896fd7cb4cb0031ba249'
Old sha1sum was: 'da39a3ee5e6b4b0d3255bfef95601890afd80709'
New sha1sum is : '4e1243bd22c66e76c2ba9eddc1f91394e57f9f83'
Old sha256sum was: 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855'
New sha256sum is : 'f2ca1bb6c7e907d06dafe4687e579fce76b37e4e93b7605022da52e6ccc26fd2'

Attributes:
 - Size: 5
 - Permissions: rw-r--r--
 - Date: Tue Dec 26 19:14:31 2023
 - Inode: 4638
 - User: root (0)
 - Group: root (0)
 - MD5: d8e8fca2dc0f896fd7cb4cb0031ba249
 - SHA1: 4e1243bd22c66e76c2ba9eddc1f91394e57f9f83
 - SHA256: f2ca1bb6c7e907d06dafe4687e579fce76b37e4e93b7605022da52e6ccc26fd2

** Alert 1703618233.25444: - ossec,syscheck,syscheck_entry_deleted,syscheck_file,pci_dss_11.5,gpg13_4.11,gdpr_II_5.1.f,hipaa_164.312.c.1,hipaa_164.312.c.2,nist_800_53_SI.7,tsc_PI1.4,tsc_PI1.5,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,
2023 Dec 26 19:17:13 (ip-172-31-33-67.ec2.internal) any->syscheck
Rule: 553 (level 7) -> 'File deleted.'
File '/etc/test1.txt' deleted
Mode: scheduled

Attributes:
 - Size: 5
 - Permissions: rw-r--r--
 - Date: Tue Dec 26 19:14:31 2023
 - Inode: 4638
 - User: root (0)
 - Group: root (0)
 - MD5: d8e8fca2dc0f896fd7cb4cb0031ba249
 - SHA1: 4e1243bd22c66e76c2ba9eddc1f91394e57f9f83
 - SHA256: f2ca1bb6c7e907d06dafe4687e579fce76b37e4e93b7605022da52e6ccc26fd2

```

</details>

<details><summary>Dashboard></summary>

![image](https://github.com/wazuh/wazuh/assets/69121070/b07de531-e295-490f-a253-35d1e9fe475e)

![image](https://github.com/wazuh/wazuh/assets/69121070/9ccf0711-c15c-4082-8095-79a94d613f1b)

</details>

</details>

###  :green_circle: FIM: Real-time

Configuration:
```
    <directories realtime="yes">/testrealtime</directories>
```

<details><summary>Results</summary>

```console
[root@ip-172-31-33-67 ec2-user]# touch /testrealtime/test1.txt
[root@ip-172-31-33-67 ec2-user]# nano /testrealtime/test1.txt
[root@ip-172-31-33-67 ec2-user]# rm /testrealtime/test1.txt
rm: remove regular file '/testrealtime/test1.txt'? yes

```

<details><summary>Dashboard</summary>

![image](https://github.com/wazuh/wazuh/assets/69121070/28a0986e-0ea3-4b33-9dea-83f528826a3a)

![image](https://github.com/wazuh/wazuh/assets/69121070/51faf51f-5734-47a6-a35f-0a3a7197c486)

</details>

<details><summary>Manager's log</summary>

```console

** Alert 1703619345.29220: - ossec,syscheck,syscheck_entry_added,syscheck_file,pci_dss_11.5,gpg13_4.11,gdpr_II_5.1.f,hipaa_164.312.c.1,hipaa_164.312.c.2,nist_800_53_SI.7,tsc_PI1.4,tsc_PI1.5,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,
2023 Dec 26 19:35:45 (ip-172-31-33-67.ec2.internal) any->syscheck
Rule: 554 (level 5) -> 'File added to the system.'
File '/testrealtime/test1.txt' added
Mode: realtime

Attributes:
 - Size: 0
 - Permissions: rw-r--r--
 - Date: Tue Dec 26 19:35:45 2023
 - Inode: 9143323
 - User: root (0)
 - Group: root (0)
 - MD5: d41d8cd98f00b204e9800998ecf8427e
 - SHA1: da39a3ee5e6b4b0d3255bfef95601890afd80709
 - SHA256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855

** Alert 1703619363.29927: - ossec,syscheck,syscheck_entry_modified,syscheck_file,pci_dss_11.5,gpg13_4.11,gdpr_II_5.1.f,hipaa_164.312.c.1,hipaa_164.312.c.2,nist_800_53_SI.7,tsc_PI1.4,tsc_PI1.5,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,
2023 Dec 26 19:36:03 (ip-172-31-33-67.ec2.internal) any->syscheck
Rule: 550 (level 7) -> 'Integrity checksum changed.'
File '/testrealtime/test1.txt' modified
Mode: realtime
Changed attributes: size,mtime,md5,sha1,sha256
Size changed from '0' to '5'
Old modification time was: '1703619345', now it is '1703619363'
Old md5sum was: 'd41d8cd98f00b204e9800998ecf8427e'
New md5sum is : 'd8e8fca2dc0f896fd7cb4cb0031ba249'
Old sha1sum was: 'da39a3ee5e6b4b0d3255bfef95601890afd80709'
New sha1sum is : '4e1243bd22c66e76c2ba9eddc1f91394e57f9f83'
Old sha256sum was: 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855'
New sha256sum is : 'f2ca1bb6c7e907d06dafe4687e579fce76b37e4e93b7605022da52e6ccc26fd2'

Attributes:
 - Size: 5
 - Permissions: rw-r--r--
 - Date: Tue Dec 26 19:36:03 2023
 - Inode: 9143323
 - User: root (0)
 - Group: root (0)
 - MD5: d8e8fca2dc0f896fd7cb4cb0031ba249
 - SHA1: 4e1243bd22c66e76c2ba9eddc1f91394e57f9f83
 - SHA256: f2ca1bb6c7e907d06dafe4687e579fce76b37e4e93b7605022da52e6ccc26fd2

** Alert 1703619370.31176: - ossec,syscheck,syscheck_entry_deleted,syscheck_file,pci_dss_11.5,gpg13_4.11,gdpr_II_5.1.f,hipaa_164.312.c.1,hipaa_164.312.c.2,nist_800_53_SI.7,tsc_PI1.4,tsc_PI1.5,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,
2023 Dec 26 19:36:10 (ip-172-31-33-67.ec2.internal) any->syscheck
Rule: 553 (level 7) -> 'File deleted.'
File '/testrealtime/test1.txt' deleted
Mode: realtime

Attributes:
 - Size: 5
 - Permissions: rw-r--r--
 - Date: Tue Dec 26 19:36:03 2023
 - Inode: 9143323
 - User: root (0)
 - Group: root (0)
 - MD5: d8e8fca2dc0f896fd7cb4cb0031ba249
 - SHA1: 4e1243bd22c66e76c2ba9eddc1f91394e57f9f83
 - SHA256: f2ca1bb6c7e907d06dafe4687e579fce76b37e4e93b7605022da52e6ccc26fd2

```

</details>

</details>

###  :red_circle: FIM whodata

Configuration:
```
    <directories whodata="yes">/testwhodata</directories>
```

<details><summary>Results</summary>

Although auditd is installed by default, Whodata fails to work:

```console

[root@ip-172-31-33-67 ec2-user]# systemctl status auditd.service
● auditd.service - Security Auditing Service
     Loaded: loaded (/usr/lib/systemd/system/auditd.service; enabled; preset: enabled)
     Active: active (running) since Tue 2023-12-26 18:26:12 UTC; 1h 27min ago
       Docs: man:auditd(8)
             https://github.com/linux-audit/audit-documentation
   Main PID: 1284 (auditd)
      Tasks: 2 (limit: 4582)
     Memory: 3.0M
        CPU: 125ms
     CGroup: /system.slice/auditd.service
             └─1284 /sbin/auditd

Dec 26 18:26:11 localhost systemd[1]: Starting auditd.service - Security Auditing Service...
Dec 26 18:26:11 localhost auditd[1284]: No plugins found, not dispatching events
Dec 26 18:26:11 localhost auditd[1284]: Init complete, auditd 3.0.6 listening for events (startup state enable)
Dec 26 18:26:12 localhost augenrules[1324]: No rules
Dec 26 18:26:12 localhost systemd[1]: Started auditd.service - Security Auditing Service.
[root@ip-172-31-33-67 ec2-user]# auditctl -v
auditctl version 3.0.6

```

When verifying auditd rules, it is confirmed that this version of Amazon Linux is affected by the issue being investigated and fixed [here](https://github.com/wazuh/wazuh/issues/19399):

```console

[root@ip-172-31-33-67 ec2-user]# auditctl -l
-a never,task


```
</details>

### :yellow_circle: SCA support


Configuration

There are no official SCA policies for Amazon Linux 2023 available at Wazuh 4.7.1. They might be available just in [version 4.8.0](https://github.com/wazuh/wazuh/issues/15462).

SCA module is tested anyway, with a custom policy.

Configuration

[This usecase](https://documentation.wazuh.com/current/user-manual/capabilities/sec-config-assessment/use-cases.html#detecting-keyword-in-a-file) is used to test SCA module.

<details><summary>Results</summary>

<details><summary>Agent's log</summary>

```console
2023/12/26 20:37:50 sca: INFO: Module started.
2023/12/26 20:37:50 sca: INFO: Loaded policy '/var/ossec/ruleset/sca/cis_amazon_linux_1.yml'
2023/12/26 20:37:50 sca: INFO: Loaded policy '/var/ossec/ruleset/sca/cis_amazon_linux_2023.yml'
2023/12/26 20:37:50 sca: INFO: Loaded policy '/var/ossec/etc/custom-sca-files/keywordcheck.yml'
2023/12/26 20:37:50 sca: INFO: Starting Security Configuration Assessment scan.
2023/12/26 20:37:50 sca: INFO: Skipping policy '/var/ossec/ruleset/sca/cis_amazon_linux_1.yml': 'Check Amazon Linux version.'
2023/12/26 20:37:51 sca: WARNING: Error found while validating policy file: '/var/ossec/ruleset/sca/cis_amazon_linux_2023.yml'. Skipping it.
2023/12/26 20:37:51 sca: INFO: Starting evaluation of policy: '/var/ossec/etc/custom-sca-files/keywordcheck.yml'
2023/12/26 20:37:54 sca: INFO: Evaluation finished for policy '/var/ossec/etc/custom-sca-files/keywordcheck.yml'
2023/12/26 20:37:54 sca: INFO: Security Configuration Assessment scan finished. Duration: 4 seconds.
2023/12/26 20:38:07 rootcheck: INFO: Ending rootcheck scan.
2023/12/26 20:38:49 wazuh-syscheckd: INFO: (6008): File integrity monitoring scan started.
2023/12/26 20:38:49 wazuh-syscheckd: INFO: (6009): File integrity monitoring scan ended.

```

</details>

<details><summary>Manager's log</summary>

```console
** Alert 1703623076.35251: - sca,gdpr_IV_35.7.d,pci_dss_2.2,nist_800_53_CM.1,tsc_CC7.1,tsc_CC7.2,
2023 Dec 26 20:37:56 (ip-172-31-33-67.ec2.internal) any->sca
Rule: 19007 (level 7) -> 'SCA use case: Keyword check: Ensure password is disabled in the test configuration file'
{"type":"check","id":1312611892,"policy":"SCA use case: Keyword check","policy_id":"keyword_check","check":{"id":10000,"title":"Ensure password is disabled in the test configuration file","description":"Password is enabled in the test configuration file.","rationale":"Password is considered weak for the custom test application. Threat actors can brute-force your password.","remediation":"Disable password by setting the value of the password_enabled option to no.","rules":["f:/usr/share/testfile.txt -> r:^password_enabled: yes$"],"condition":"none","file":"/usr/share/testfile.txt","result":"failed"}}
sca.type: check
sca.scan_id: 1312611892
sca.policy: SCA use case: Keyword check
sca.check.id: 10000
sca.check.title: Ensure password is disabled in the test configuration file
sca.check.description: Password is enabled in the test configuration file.
sca.check.rationale: Password is considered weak for the custom test application. Threat actors can brute-force your password.
sca.check.remediation: Disable password by setting the value of the password_enabled option to no.
sca.check.file: ["/usr/share/testfile.txt"]
sca.check.result: failed

** Alert 1703623083.36679: - sca,gdpr_IV_35.7.d,pci_dss_2.2,nist_800_53_CM.1,tsc_CC7.1,tsc_CC7.2,
2023 Dec 26 20:38:03 (ip-172-31-33-67.ec2.internal) any->sca
Rule: 19005 (level 9) -> 'SCA summary: SCA use case: Keyword check: Score less than 30% (0)'
{"type":"summary","scan_id":1312611892,"name":"SCA use case: Keyword check","policy_id":"keyword_check","file":"keywordcheck.yml","description":"Guidance for checking for a keyword or phrase in files on Ubuntu endpoints.","references":"https://documentation.wazuh.com/current/user-manual/capabilities/sec-config-assessment/index.html,https://documentation.wazuh.com/current/user-manual/capabilities/sec-config-assessment/creating-custom-policies.html","passed":0,"failed":1,"invalid":0,"total_checks":1,"score":0,"start_time":1703623071,"end_time":1703623071,"hash":"5d28a90f4498a81461efbaf6f628a19d9778390bb5c81a393dd936181cc3d826","hash_file":"a33741eefba60ba75052477564b607cc97bdf36c773194c7ead40f30e356ec4d","force_alert":"1"}
sca.type: summary
sca.scan_id: 1312611892
sca.policy: SCA use case: Keyword check
sca.description: Guidance for checking for a keyword or phrase in files on Ubuntu endpoints.
sca.policy_id: keyword_check
sca.passed: 0
sca.failed: 1
sca.invalid: 0
sca.total_checks: 1
sca.score: 0
sca.file: keywordcheck.yml

** Alert 1703623092.37970: - sca,gdpr_IV_35.7.d,pci_dss_2.2,nist_800_53_CM.1,tsc_CC7.1,tsc_CC7.2,
2023 Dec 26 20:38:12 (ip-172-31-33-67.ec2.internal) any->sca
Rule: 19005 (level 9) -> 'SCA summary: SCA use case: Keyword check: Score less than 30% (0)'
{"type":"summary","scan_id":1312611892,"name":"SCA use case: Keyword check","policy_id":"keyword_check","file":"keywordcheck.yml","description":"Guidance for checking for a keyword or phrase in files on Ubuntu endpoints.","references":"https://documentation.wazuh.com/current/user-manual/capabilities/sec-config-assessment/index.html,https://documentation.wazuh.com/current/user-manual/capabilities/sec-config-assessment/creating-custom-policies.html","passed":0,"failed":1,"invalid":0,"total_checks":1,"score":0,"start_time":1703623071,"end_time":1703623071,"hash":"5d28a90f4498a81461efbaf6f628a19d9778390bb5c81a393dd936181cc3d826","hash_file":"a33741eefba60ba75052477564b607cc97bdf36c773194c7ead40f30e356ec4d","force_alert":"1","force_alert":"1"}
sca.type: summary
sca.scan_id: 1312611892
sca.policy: SCA use case: Keyword check
sca.description: Guidance for checking for a keyword or phrase in files on Ubuntu endpoints.
sca.policy_id: keyword_check
sca.passed: 0
sca.failed: 1
sca.invalid: 0
sca.total_checks: 1
sca.score: 0
sca.file: keywordcheck.yml


```

</details>

<details><summary>Dashboard</summary>

![image](https://github.com/wazuh/wazuh/assets/69121070/2d18063c-45f6-48be-b1fe-a662dad0ab9f)

![image](https://github.com/wazuh/wazuh/assets/69121070/3911f11f-73f2-4818-8838-ab64cea8da42)

</details>

</details>


### :green_circle: Syscollector

Configuration by default

<details><summary>Results</summary>

<details><summary>Agent's log</summary>

```console
2023/12/26 20:37:45 wazuh-modulesd:syscollector: INFO: Module finished.
2023/12/26 20:37:50 wazuh-modulesd:syscollector: INFO: Module started.
2023/12/26 20:37:50 wazuh-modulesd:syscollector: INFO: Starting evaluation.
2023/12/26 20:37:51 wazuh-modulesd:syscollector: INFO: Evaluation finished.
```

</details>

<details><summary>Manager's log</summary>

```console
2023/12/22 18:41:48 wazuh-modulesd:syscollector: INFO: Module started.
2023/12/22 18:41:48 wazuh-modulesd:syscollector: INFO: Starting evaluation.
2023/12/22 18:41:48 wazuh-modulesd:syscollector: INFO: Evaluation finished.
```

</details>

<details><summary>Inventory</summary>

![image](https://github.com/wazuh/wazuh/assets/69121070/ec0984f3-d2e6-45b9-bb05-a36b0feaf707)

![image](https://github.com/wazuh/wazuh/assets/69121070/6dc00515-08d4-4b49-84fd-334344acee2d)

![image](https://github.com/wazuh/wazuh/assets/69121070/9d971660-6829-45e4-a0af-ac2c9f062277)

</details>

</details>

### :green_circle: Log capture (Eventchannel for Windows and ULS for macOS)

A log file will be traced and a log entry will be added to verify it is successfully detected

Agent's configuration:

```console

  <localfile>
    <log_format>syslog</log_format>
    <location>/var/log/file.log</location>
  </localfile>

```


We force an event in the file:

```console

echo 'Sep 10 22:12:41 example pvedaemon[6427]: authentication failure; rhost=192.168.0.1 user=root@pam msg=Authentication failure' >>/var/log/file.log

```
<details><summary>Results</summary>

<details><summary>Manager's log</summary>

```console
** Alert 1703638814.1535: - syslog,proxmox-ve,authentication_failed,pci_dss_10.2.4,pci_dss_10.2.5,gdpr_IV_35.7.d,gdpr_IV_32.2,hipaa_164.312.b,nist_800_53_AU.14,nist_800_53_AC.7,tsc_CC6.1,tsc_CC6.8,tsc_CC7.2,tsc_CC7.3,
2023 Dec 27 01:00:14 (ip-172-31-33-67.ec2.internal) any->/var/log/file.log
Rule: 87201 (level 6) -> 'Proxmox VE authentication failed.'
Src IP: 192.168.0.1
User: root
Sep 10 22:12:41 example pvedaemon[6427]: authentication failure; rhost=192.168.0.1 user=root@pam msg=Authentication failure
```

</details>

<details><summary>Dashboard</summary>

![image](https://github.com/wazuh/internal-devel-requests/assets/69121070/8a3b594d-3858-437c-a95b-d10134eb0bbc)

</details>

</details>

**Comment by lchico** [comment]
LGTM! Thanks @ncvicchi !
